atom/browser/auto_updater_mac.mm

```diff
@@ -29,35 +29,40 @@
 }
 
 // static
-void AutoUpdater::SetFeedURL(const std::string& feed) {
-  if (g_updater == nil) {
-    Delegate* delegate = GetDelegate();
-    if (!delegate)
-      return;
-
-    // Initialize the SQRLUpdater.
-    NSURL* url = [NSURL URLWithString:base::SysUTF8ToNSString(feed)];
-    NSURLRequest* urlRequest = [NSURLRequest requestWithURL:url];
-
-    @try {
-      g_updater = [[SQRLUpdater alloc] initWithUpdateRequest:urlRequest];
-    } @catch (NSException* error) {
-      delegate->OnError(base::SysNSStringToUTF8(error.reason));
-      return;
-    }
-
-    [[g_updater rac_valuesForKeyPath:@"state" observer:g_updater]
-      subscribeNext:^(NSNumber *stateNumber) {
-        int state = [stateNumber integerValue];
-        // Dispatching the event on main thread.
-        dispatch_async(dispatch_get_main_queue(), ^{
-          if (state == SQRLUpdaterStateCheckingForUpdate)
-            delegate->OnCheckingForUpdate();
-          else if (state == SQRLUpdaterStateDownloadingUpdate)
-            delegate->OnUpdateAvailable();
-        });
-    }];
+void AutoUpdater::SetFeedURL(const std::string& feed, const std::string& authToken) {
```

Comment:
Add comments for function header changes and inside setFeedURL
Make setFeedURL overloaded for back compat
null check on authtoken.